src/ts/ssh/progress.ts

```diff
@@ -0,0 +1,39 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//
+
+/**
+ * Connection progress
+ */
+export enum Progress {
+	// Client connection to relay
+	OpeningClientConnectionToRelay = "OpeningClientConnectionToRelay",
+	OpenedClientConnectionToRelay = "OpenedClientConnectionToRelay",
+
+	// Host connection to relay
+	OpeningHostConnectionToRelay = "OpeningHostConnectionToRelay",
+	OpenedHostConnectionToRelay = "OpenedHostConnectionToRelay",
+
+	// SSH Session Connection
+	OpeningSshSessionConnection = "OpeningSshSessionConnection",
+	OpenedSshSessionConnection = "OpenedSshSessionConnection",
+	SendingProtocolVersionExchange = "SendingProtocolVersionExchange",
+	ReceivingProtocolVersionExchange = "ReceivingProtocolVersionExchange",
+	StartingKeyExchange = "StartingKeyExchange",
+	CompletedKeyExchange = "CompletedKeyExchange",
+	StartingSessionAuthentication = "StartingSessionAuthentication",
+	CompletedSessionAuthentication = "CompletedSessionAuthentication",
+
+	// Port forwarding messages
+	StartingPortForwarding = "StartingPortForwarding",
+	CompletedLocalPortForwarding = "CompletedLocalPortForwarding",
+	CompletedRemotePortForwarding = "CompletedRemotePortForwarding",
+	OpeningChannelPortForwarding = "OpeningChannelPortForwarding",
+	OpenedChannelPortForwarding = "OpenedChannelPortForwarding",
```

Comment:
There are already `PortForwarding` and `ChannelOpening` events which have more details than these progress events. Do you think the progress events are still useful? Or should applications listen to progress events for only basic session connection operations and use the other events for port-forwarding and channel opening?

Reply:
If there is already more detailed events available, I'll remove them from here. If we find folks are losing the connection thread at that point then we can add them back in later.